src/components/Form.jsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+
+import "./TodoForm.css";
+
+Form.propTypes = {
+  onSubmit: PropTypes.func.isRequired,
+  initialValue: PropTypes.string,
+  placeholder: PropTypes.string,
+  buttonText: PropTypes.string,
+};
+
+export function Form({ onSubmit, initialValue, placeholder, buttonText }) {
+  const [value, setValue] = useState(initialValue || "");
+  const [hasError, setHasError] = useState(false);
+
+  const handleSubmit = (e) => {
+    if (value.trim() !== "") {
+      onSubmit(value);
+      setValue("");
+      setHasError(false);
+    } else {
+      setHasError(true);
+    }
+  };
+
+  return (
+    <div className="TodoForm" onSubmit={handleSubmit} role='form'>
```

Comment:
note this pattern (prob. you'll see this in the future).

HTML's forms have default behaviour, (that's why you needed those `stopPropagation()`) one pattern that is widely used is: pretend that a regular `<div>` is a form. As this example. 

If you're curious the role thing is related to a11y (acronym for accesibility).